article/_posts/2019-06-27-RPL.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: review
+title: "Residual Policy Learning"
+tags: deep-learning reinforcement
+author: "Antoine Théberge"
+cite:
+    authors: "Tom Silver, Kelsey Allen, Josh Tenenbaum, Leslie Kaelbling"
+    title:   "Residual Policy Learning"
+    venue:   "Arxiv 2018"
+pdf: "https://arxiv.org/abs/1812.06298"
+---
+
+
+# Highlights
+- RPL improves on initial policies
+- RPL is more data-efficient than learning from scratch
+
+# Introduction
+
+For most complex problems of interest, learning from scratch remains intractable, especially in long-horizon sparse-reward environments. Often, it is easier to provide a hand-crafted policy or a model-predictive controller in the case of robotics. But, these policies are often barely _good enough_ and don't do well in hard-to-model environments (involving friction for example) or environments full of obstacles.
+
+In this work, the authors propose _Residual Policy Learning_ (RPL), a method for improving existing policies.
+
+# Methods
+
+![](/article/images/RPL/push.jpeg)
+
+The main idea is to learn to augment existing policies by learning _residuals_ on top of them. Given a policy $$\pi:S \rightarrow A$$, the authors propose to learn a residual function $$f_\theta(s):S \rightarrow A$$ given by
+
+$$\pi_\theta(s) = \pi(s) + f_\theta(s)$$
+
+Because the policy $$\pi$$ is fixed, the gradient of the policy $$\pi_\theta$$ is given by:
+
+$$\nabla_{\theta}\pi_{\theta}(s) = \nabla_{\theta}f_{\theta}(s)$$
+
+Therefore, RPL can be used to improve on non-differentiable policies.
+
+To use these two policies together, let's remind ourselves that the policy $$\pi$$ acts in the MDP $$M = (S,A, R, T, \gamma)$$ where $$S$$ is the set of all possible states, $$A$$ the set of all possible actions, $$R$$ the reward for taking action $$a \in A$$ at $$s \in S$$, $$T(s, a, s')$$ the transition probabilities and $$\gamma$$ the discount factor.
+
+The residual policy $$f_\theta$$ then acts in the _residual MDP_ $$M^{(\pi)} = (S,A, R, T^{(\pi)}, \gamma)$$ where
+
+$$T^{(\pi)}(s, a, s') = T(s, \pi(s)+a,s')$$
+
+Thus, standard reinforcement learning can be used to learn the residual policy.
+
+![](/article/images/RPL/fig5.jpeg)
+
+## Initializing the residuals
+
+One thing to make sure is that the residuals don't make the initial policy worse. If the initial policy is _perfect_, the residuals should have no influence. To alleviate this problem, the authors initialize the residual function so that $$f_{\theta}(s) = 0$$ by making the weights of the last layer of their network all zero.
+
+## RPL with Actor-Critic Methods
+
+Actor-Critic method where the gradient of the actor depends on the critic and a bad critic may make the actor _worse_. The authors, therefore, propose to train the critic alone first until its loss dips below a threshold $$\Beta$$, which they fixed at $$\Beta = 1.0$$
```

Comment:
`Actor-Critic method where the gradient of the actor depends on the critic and a bad critic may make the actor _worse_.`
The sentence seems to be missing the verb/unfinished.

Reply:
Woops, fixed in "99802cd"